COMMUNICATIONS.md

```diff
@@ -1,3 +1,199 @@
+mind. If you decide to analyze someone's technical work or public
+statements more closely because you see their associations, so be
+it. But it is inappropriate to discredit their work simply based on
+these prejudices.
+
+Find a technical issue to discuss, or leave the matter alone.
+
+### Encourage healthy discussion
+
+We _want_ open, healthy discussion around our projects. Since this is
+about Stack: user feedback has dramatically improved Stack over the
+years. Even negative feedback from non-users has in many cases helped
+Stack improve. These are good things worth discussing.
+
+Do not try to shut down or avoid a healthy discussion.
+
+### Debate technical merits
+
+Discussions should be about the technical merits of a
+project. However, the definition of technical merits is in fact quite
+broad, including:
+
+* Performance
+* Reliability
+* Maintainability
+* Ease of use
+* Marketability
+
+I believe that last one may be a surprise to many. I want to make the
+argument for that specifically. The best software is useless (by
+definition) if no one uses it. Making an argument for a feature so
+that it is easier to encourage the use of some software is perfectly
+valid.
+
+That said, keep these discussions as objective as possible.
+
+### Avoid partisanship
+
+This document is written for the Stack community. As [the initial
+proposal](https://www.snoyman.com/blog/2018/11/proposal-stack-coc)
+mentions, this in itself is not ideal, since it is somewhat
+divisive. Allow me to backtrack that divisiveness here a bit: we
+should _not_ be trying to establish a separate Stack community that
+does not integrate with other projects.
+
+Much of how Stack has been designed is in fact along these lines:
+
+* The Stackage project is separate, and is usable by both Nix and (to
+  some extent) cabal-install (the build tool)
+* Hackage remains the upstream repository
+* Cabal (the library, file format, and build system) remain the
+  underlying mechanism for representing and building packages
+
+Broadening this a bit: I (Michael) am one of the original authors of
+Stack. I'm also the original author of Yesod and Conduit. It is
+absolutely vital that Stack treat other web frameworks (like Servant)
+or streaming data libraries (like Pipes or Streaming) as first class
+citizens, in exactly the same way as Yesod or Conduit.
+
+Said another way: a decision to opt in to one piece of the toolchain
+should be orthogonal to any other decisions.
+
+### Respect others' time
+
+This is covered in depth in the "prior art" links above. Most people
+are interacting with the Stack community in their free time. Companies
+that support development of tooling are spending funds they could be
+using for non-community activities. We should respect these
+contributions.
+
+On the other hand, for users and consumers of the technology, we
+should wherever possible try to make life as easy as we can. Avoid
+blocking people if you can. Recognize differences in goals and see if
```

Comment:
```suggestion
creating blockers for people if you can. Recognize differences in goals and see if
```

Reply:
Phrasing -- "Avoid blocking people" brings to mind "blocking as in Twitter", as opposed to "creating a technical obstacle".

Reply:
Ooh, really nice catch!